File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/PackageMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.packager.ContextPackager;
import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import java.util.concurrent.CompletableFuture;

public class PackageMiddleware implements PipelineMiddleware {
    private final ContextPackager p;
    public PackageMiddleware(ContextPackager cp) { this.p = cp; }
    public String name() { return "package"; }
    public int order() { return 500; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        ctx.setPackagedContext(p.pack(ctx.optimizedContext(), ctx.provider()));
        return CompletableFuture.completedFuture(null);
    }
}

File: context_os/pipeline/middlewares/package_middleware.py
"""PackageMiddleware — Prompt 打包。"""

from context_os.packager.packager import ContextPackager
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware


class PackageMiddleware(PipelineMiddleware):
    """按 provider 格式拼接 sections → PackagedContext。"""

    def __init__(self, packager: ContextPackager):
        self._p = packager

    def name(self) -> str:
        return "package"

    def order(self) -> int:
        return 500

    async def execute(self, ctx: PipelineContext) -> None:
        ctx.packaged_context = self._p.pack(ctx.optimized_context, ctx.provider)
